lib/test_boosters/leftover_specs.rb

```diff
@@ -0,0 +1,26 @@
+module LeftoverSpecs
+  def self.select(all_leftover_specs, thread_count, thread_index)
+    all_leftover_specs = sort_by_size(all_leftover_specs)
+
+    return [] if all_leftover_specs.empty?
+
+    specs = all_leftover_specs
+      .each_slice(thread_count)
+      .reduce{|acc, slice| acc.map{|a| a}.zip(slice.reverse)}
+      .map{|f| if f.kind_of?(Array) then f.flatten else [f] end} [thread_index]
+
+    if    specs.nil?            then []
+    elsif specs.kind_of?(Array) then specs.compact
+    end
+  end
+
+  def self.sort_by_size(specs) # descending
+    specs
+      .map{|f| if File.file?(f) then f else nil end}
+      .compact
```

Comment:
You can replace these two lines with `.select { |f| File.file?(f) }`.

Also, it is customary to write `[].map { |x| x }` instead of `[].map{|x| x}`.